lib/diaspora/federated/shareable.rb

```diff
@@ -101,18 +101,18 @@ def update_existing_sharable(shareable)
           end
         end
 
-        def receive_shareable_visibility(user, person, shareable)
-          user.contact_for(person).receive_shareable(shareable)
+        def receive_shareable_visibility(user, shareable)
+          user.receive_shareable(shareable)
           user.notify_if_mentioned(shareable)
           logger.info "event=receive payload_type=#{self.class} status=complete " \
-                      "sender=#{diaspora_handle} receiver=#{person.diaspora_handle} guid=#{shareable.guid}"
+                      "sender=#{diaspora_handle} receiver=#{user.diaspora_handle} guid=#{shareable.guid}"
         end
 
-        def receive_non_persisted(user, person)
+        def receive_non_persisted(user)
```

Comment:
Assignment Branch Condition size for receive_non_persisted is too high. [23.43/20]